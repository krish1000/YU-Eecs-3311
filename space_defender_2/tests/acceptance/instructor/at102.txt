--Test Spread collision detection is different from Starfighter moving for diagonal moving projectiles.
play(10,10,1,1,1,1,1)
setup_select(2)
setup_next(5)

toggle_debug_mode


move(D,7)
fire
move(A,9)
pass

move(G,7)
fire
move(J,9)
move(J,8)
move(J,9)
