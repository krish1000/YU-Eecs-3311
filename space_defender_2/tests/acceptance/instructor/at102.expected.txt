  state:not started, normal, ok
  Welcome to Space Defender Version 2.
->play(10,10,1,1,1,1,1)
  state:weapon setup, normal, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_select(2)
  state:weapon setup, normal, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Spread
->setup_next(5)
  state:in game(0.0), normal, ok
  Starfighter:
    [0,S]->health:60/60, energy:120/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[E,1]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  ?
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E S  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  ?
    J _  _  _  _  _  _  _  _  ?  ?
->toggle_debug_mode
  state:in game(0.1), debug, ok
  In debug mode.
->move(D,7)
  state:in game(1.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:92/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[D,7]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [E,1] -> [D,7]
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  S  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(2.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:86/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[D,7]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
    [-1,*]->damage:50, move:1, location:[C,8]
    [-2,*]->damage:50, move:1, location:[D,8]
    [-3,*]->damage:50, move:1, location:[E,8]
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [D,7].
      A friendly projectile(id:-1) spawns at location [C,8].
      A friendly projectile(id:-2) spawns at location [D,8].
      A friendly projectile(id:-3) spawns at location [E,8].
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  *  _  _
    D _  _  _  _  _  _  S  *  _  _
    E _  _  _  _  _  _  _  *  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->move(A,9)
  state:in game(3.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:70/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[A,9]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
    [-1,*]->damage:50, move:1, location:[B,9]
    [-2,*]->damage:50, move:1, location:[D,9]
    [-3,*]->damage:50, move:1, location:[F,9]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [C,8] -> [B,9]
    A friendly projectile(id:-2) moves: [D,8] -> [D,9]
    A friendly projectile(id:-3) moves: [E,8] -> [F,9]
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [D,7] -> [A,9]
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  S  _
    B _  _  _  _  _  _  _  _  *  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  *  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  *  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->pass
  state:in game(4.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:78/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[A,9]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
    [-1,*]->damage:50, move:1, location:[A,10]
    [-2,*]->damage:50, move:1, location:[D,10]
    [-3,*]->damage:50, move:1, location:[G,10]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [B,9] -> [A,10]
    A friendly projectile(id:-2) moves: [D,9] -> [D,10]
    A friendly projectile(id:-3) moves: [F,9] -> [G,10]
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) passes at location [A,9], doubling regen rate.
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  S  *
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  *
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  *
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->move(G,7)
  state:in game(5.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:50/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[G,7]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [A,10] -> out of board
    A friendly projectile(id:-2) moves: [D,10] -> out of board
    A friendly projectile(id:-3) moves: [G,10] -> out of board
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [A,9] -> [G,7]
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  S  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(6.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:44/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[G,7]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
    [-4,*]->damage:50, move:1, location:[F,8]
    [-5,*]->damage:50, move:1, location:[G,8]
    [-6,*]->damage:50, move:1, location:[H,8]
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [G,7].
      A friendly projectile(id:-4) spawns at location [F,8].
      A friendly projectile(id:-5) spawns at location [G,8].
      A friendly projectile(id:-6) spawns at location [H,8].
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  *  _  _
    G _  _  _  _  _  _  S  *  _  _
    H _  _  _  _  _  _  _  *  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  _  _
->move(J,9)
  state:in game(7.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:28/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[J,9]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
    [-4,*]->damage:50, move:1, location:[E,9]
    [-5,*]->damage:50, move:1, location:[G,9]
    [-6,*]->damage:50, move:1, location:[I,9]
  Friendly Projectile Action:
    A friendly projectile(id:-4) moves: [F,8] -> [E,9]
    A friendly projectile(id:-5) moves: [G,8] -> [G,9]
    A friendly projectile(id:-6) moves: [H,8] -> [I,9]
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [G,7] -> [J,9]
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  *  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  *  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  *  _
    J _  _  _  _  _  _  _  _  S  _
->move(J,8)
  state:in game(8.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:28/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[J,8]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
    [-4,*]->damage:50, move:1, location:[D,10]
    [-5,*]->damage:50, move:1, location:[G,10]
    [-6,*]->damage:50, move:1, location:[J,10]
  Friendly Projectile Action:
    A friendly projectile(id:-4) moves: [E,9] -> [D,10]
    A friendly projectile(id:-5) moves: [G,9] -> [G,10]
    A friendly projectile(id:-6) moves: [I,9] -> [J,10]
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [J,9] -> [J,8]
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  *
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  *
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  S  _  *
->move(J,9)
  state:in game(9.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:28/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[J,9]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
    A friendly projectile(id:-4) moves: [D,10] -> out of board
    A friendly projectile(id:-5) moves: [G,10] -> out of board
    A friendly projectile(id:-6) moves: [J,10] -> out of board
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [J,8] -> [J,9]
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
    F _  _  _  _  _  _  _  _  _  _
    G _  _  _  _  _  _  _  _  _  _
    H _  _  _  _  _  _  _  _  _  _
    I _  _  _  _  _  _  _  _  _  _
    J _  _  _  _  _  _  _  _  S  _
